clients/privacy-center/features/consent/hooks.ts

```diff
@@ -0,0 +1,43 @@
+import { getOrMakeFidesCookie } from "fides-js";
+import { useEffect } from "react";
+import { useAppDispatch, useAppSelector } from "~/app/hooks";
+import { PrivacyNoticeRegion } from "~/types/api";
+import { selectIsNoticeDriven } from "~/features/common/settings.slice";
+import {
+  useGetPrivacyExperienceQuery,
+  setRegion,
+  setFidesDeviceUserId,
+  selectExperienceRegion,
+} from "./consent.slice";
+
+/**
+ * Subscribes to the relevant privacy experience, passing in parameters
+ * for the user's region and their fides_user_device_id. By calling this,
+ * the selector for experiences should then be populated.
+ *
+ * const experience = useAppSelector(selectPrivacyExperience);
+ *
+ * Skips the subscription if notices are not enabled via settings.
+ */
+export const useSubscribeToPrivacyExperienceQuery = () => {
+  const skip = !useAppSelector(selectIsNoticeDriven);
+  const dispatch = useAppDispatch();
+  const cookie = getOrMakeFidesCookie();
+  const { fides_user_device_id: fidesUserDeviceId } = cookie.identity;
+
+  useEffect(() => {
+    // TODO: query for location
```

Comment:
How should we query for location here? import `getGeolocation` from `fides-js`, but only if the option is enabled? 

What happens if we don't get a well formatted region back? I guess there would be no experience then?

Reply:
okay, this is added here: https://github.com/ethyca/fides/pull/3340/commits/c9dd8270f99677c42708bf485caf4acb045986a8

it seems to work, though I have to use my vpn to switch locations when manually testing. I'm not sure if the privacy center can "inject" a geolocation the same way that fides-js can? we can stub the location response via Cypress at least

Reply:
okay, this is added here: https://github.com/ethyca/fides/pull/3340/commits/c9dd8270f99677c42708bf485caf4acb045986a8

it seems to work, though I have to use my vpn to switch locations when manually testing. I'm not sure if the privacy center can "inject" a geolocation the same way that fides-js can? we can stub the location response via Cypress at least